_ingest-pipelines/processors/join.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: default
+title: Join
+parent: Ingest processors
+nav_order: 160
+---
+
+# Join processor
+
+The `join` processor combines fields from different data sources into a single document before indexing. Raises an error if the field is not an array. For example, you could combine the log message, severity level, and timestamp into a single field for better readability and easier querying. Or, you could join data from different sources, such as application logs and system logs, based on a common field like a session ID or user ID to provide a more comprehensive view of related events and help in troubleshooting and root cause analysis. 
+
+The following is the syntax for the `join` processor:
+
+```json
+{
+  "join": {
+    "field": "field_name",
+    "separator": "separator_string"
+  }
+}
+```
+{% include copy-curl.html %}
+
+## Configuration parameters
+
+The following table lists the required and optional parameters for the `join` processor.
+
+Parameter | Required/Optional | Description |
+|-----------|-----------|-----------|
+`field` | Required | The field name where the join operator is applied. 
```

Comment:
`field` must be an array, we should emphasize this.

Reply:
Revised